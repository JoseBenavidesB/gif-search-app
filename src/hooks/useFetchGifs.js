import { useEffect, useState } from "react"
import { getGifts } from "../helpers/getGifts";



export const useFetchGifs = ( category ) => {
    
    const [state, setState ] = useState({
        data: [],
        loading: true
    });

    useEffect( ()=> {
        
        getGifts(category)
            .then(gifs => setState({
                data: gifs,
                loading: false
            }))
    }, [category]); //d[deoendencies, in this case: category. when category change, this load again] */

    return state;
}
