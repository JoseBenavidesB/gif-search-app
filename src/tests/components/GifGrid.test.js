import React from 'react'
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom'

jest.mock('../../hooks/useFetchGifs'); // fingir

describe('Test <GifGrid/>', () => { 

    const category = 'One punch';

    test('should return <GifGrid/> perfectly', ()=> {
    
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show items when load img with useFetchGifs', () => { 
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost.com',
            title: 'Test title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    
    
})