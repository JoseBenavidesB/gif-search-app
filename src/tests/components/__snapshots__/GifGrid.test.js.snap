// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid/> should return <GifGrid/> perfectly 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="group"
  />
</Fragment>
`;

exports[`Test <GifGrid/> should show items when load img with useFetchGifs 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="group"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Test title"
      url="https://localhost.com"
    />
  </div>
</Fragment>
`;
